src/components/transport_manager/test/transport_manager_impl_test.cc

```diff
@@ -101,11 +106,40 @@ class TransportManagerImplTest : public ::testing::Test {
         MakeShared<RawMessage>(connection_key_, version_protocol_, data, kSize);
   }
 
-  void HandleDeviceListUpdated() {
-    const int type = static_cast<int>(
-        TransportAdapterListenerImpl::EventTypeEnum::ON_DEVICE_LIST_UPDATED);
+  DeviceInfo ConstructDeviceInfo(const std::string& mac_address,
+                                 const std::string& connection_type,
+                                 const std::string& device_name) {
+    const auto device_handle(
+        tm_.get_converter().UidToHandle(mac_address, connection_type));
+
+    return DeviceInfo(device_handle, mac_address, device_name, connection_type);
+  }
+
+  void SetOnDeviceExpectations(const DeviceInfo& device_info) {
+    EXPECT_CALL(*tm_listener_, OnDeviceAdded(device_info));
+    EXPECT_CALL(*tm_listener_, OnDeviceListUpdated(_));
+  }
```

Comment:
In addition, these three expectations could be added to this common function instead of being repeated:
```
void SetDeviceExpectations(MockTransportAdapter* mock_adapter,
                                                const DeviceList& device_list,
                                                const DeviceInfo& device_info) {
   EXPECT_CALL(*mock_adapter, GetDeviceList())
       .WillRepeatedly(Return(device_list));
 
   EXPECT_CALL(*mock_adapter, DeviceName(device_info.mac_address()))
       .WillRepeatedly(Return(device_info.name()));
 
   EXPECT_CALL(*mock_adapter, GetConnectionType())
       .WillRepeatedly(Return(device_info.connection_type()));
...
```

On the other hand, you could also just use the existing `SetAddDeviceExpectations` function wherever this function would be used. The only issue with this approach is that there is an extra expectation (`GetDeviceType`) which is not used in most cases.

Reply:
@jacobkeeler Right. This is the only issue. But it is critical in this case. In case of extra expectation - unit test will fail.